modules/distr-tracing-otel-config-collector.adoc

```diff
@@ -239,6 +239,60 @@ The Zipkin receiver ingests data in the Zipkin v1 and v2 formats.
 [id="processors_{context}"]
 === Processors
 
+[id="memorylimiter-processor_{context}"]
+==== Memory Limiter processor
+
+The Memory Limiter processor periodically checks the memory usage of the collector and pauses processing of data if the soft memory limit is reached.
+The previous component (usually a receiver) is expected to retry sending the same data, and may apply a backpressure to the incoming data.
+When the memory usage is above the hard limit, the Memory Limiter processor forces garbage collection to run.
+
+* Support level: General Availability
+* Supported signals: traces, metrics, logs
+
+.Example of the OpenTelemetry Collector custom resource when using the Memory Limiter processor
+[source,yaml]
+----
+  config: |
+    processor:
+      memory_limiter:
+        check_interval: 1s
+        limit_mib: 4000
+        spike_limit_mib: 800
+    service:
+      pipelines:
+        traces:
+          processors: [batch]
+        metrics:
+          processors: [batch]
+----
+
+.Parameters used by the Memory Limiter processor
+[cols="3",options="header"]
+|===
+| Parameter | Description | Default
+
+| `check_interval`
+| Time between memory usage measurements. For spiky traffic patterns, decrease `check_interval` or increase `spike_limit_mib`. Recommendation: 1s.
+| 0s
+
+| `limit_mib`
+| Maximum amount (hard limit) of memory in MiB allocated on the heap. Typically the total memory usage of the OpenTelemetry collector will be about 50 MiB higher than this value.
+| 0
+
+| `spike_limit_mib`
+| Maximum expected spikes of memory usage. This defines the soft limit, which is calculated as `limit_mib - spike_limit_mib`. The recommendation is about 20% of `limit_mib`.
```

Comment:
Could you please explain a bit more in a reply comment below in this thread: what "the maximum expected spikes of memory usage" is?
Is it "the maximum expected [what? amount? value? of?] spikes of memory usage"?
```suggestion
| The soft limit, which is the maximum expected spikes of memory usage. This is calculated as `limit_mib - spike_limit_mib`. The optimal value is about 20% of `limit_mib`.
```

Reply:
After you clarify the above, I can update my suggestion below:
```suggestion
| The soft limit, which is the maximum expected spikes of memory usage. This is calculated as `limit_mib - spike_limit_mib`. The optimal value is about 20% of `limit_mib`.
```

Reply:
For example the average memory usage of the collector is 400 MiB, and once in a while some process sends a big amount of metrics/logs/traces in one go and then for a short time period the collector uses 500 MiB because it has to queue up the incoming data, because it cannot process everything immediately. In this case, the spike was 100 MiB.

Then we should use `limit_mib: 500` (the hard limit) and `spike_limit_mib: 100` (the size in MiB of the memory spike). Now, if the memory usage is between 400 (soft limit) and 500 (hard limit), the collector pauses receiving any new data until the backlog is completed and the memory usage is under the soft limit again.

Reply:
Thank you for clarifying this, Andreas.

Reply:
`spike_limit_mib` is not the soft limit. It's the maximum expected memory spike in MiB. With this value you can calculate the soft limit, by using the formula `soft_limit = limit_mib - spike_limit_mib`.

I think the options are confusing (imho it would be much simpler to have an option called `soft_limit_mib` or something and calculate the spike amount in the code), but we can't change that in the docs (unless we change the source code).

Maybe
```
| Maximum expected memory spike in MiB. This defines the soft limit, which ...
```
? Not sure if there's a better way to describe `spike`. Here are the upstream docs: https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiterprocessor/README.md

Reply:
I'm trying to understand where or what for the `soft_limit` is needed. Does the reader need to know the value of the `soft_limit`? Like for entering it somewhere else?

Reply:
Yes, the soft limit is important for the reader to know. It doesn't need to be entered (it's calculated from the configuration options with the formula above), and is used by the component:

>  [...] and **pauses data processing when the soft memory limit is reached**.

(from the description)

Reply:
Thank you, Andreas. What do you think about my latest suggestion? https://github.com/openshift/openshift-docs/pull/65036/files#r1350518182